### PR TITLE
Target .NET Core 2.0 and .NET Framework 4.6.1

### PR DESCRIPTION
**Pull request by natemcmaster**
Removes .NET Standard support.

See https://github.com/aspnet/EventNotification/pull/70#issuecomment-307203075

### CODE REVIEW

File: src/Microsoft.Extensions.DiagnosticAdapter/DiagnosticListenerExtensions.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public static IDisposable SubscribeWithAdapter(this DiagnosticListener diagnosti
             Func<string, object, object, bool> isEnabled)
         {
             var adapter = new DiagnosticSourceAdapter(target, isEnabled);
-            return diagnostic.Subscribe(adapter, adapter.IsEnabled);
+            return diagnostic.Subscribe(adapter, (Predicate<string>)adapter.IsEnabled);
```
Comment (rynowak) [inline src/Microsoft.Extensions.DiagnosticAdapter/DiagnosticListenerExtensions.cs]: Why are these changes required?

File: src/Microsoft.Extensions.DiagnosticAdapter/Internal/ProxyTypeEmitter.cs [hunk 2]
```diff
@@ -154,7 +154,7 @@ internal static bool VerifyProxySupport(ProxyBuilderContext context, Tuple<Type,
                     VerificationResult elementResult;
                     context.Visited.TryGetValue(elementKey, out elementResult);
 
-                    var proxyType = elementResult?.Type?.GetTypeInfo() ?? (TypeInfo)elementResult?.TypeBuilder;
+                    var proxyType = elementResult?.Type ?? elementResult?.TypeBuilder as Type;
```
Comment (rynowak) [inline src/Microsoft.Extensions.DiagnosticAdapter/Internal/ProxyTypeEmitter.cs]: This should be a c-style cast, it's never expected to fail

File: src/Microsoft.Extensions.DiagnosticAdapter/DiagnosticListenerExtensions.cs (hunk 1 above)
Comment (natemcmaster) [inline src/Microsoft.Extensions.DiagnosticAdapter/DiagnosticListenerExtensions.cs]: .Subscribe has overloads in netcoreapp2.0/net461 that were not in ns1.3. The call was ambiguous.

Comment (rynowak) [inline src/Microsoft.Extensions.DiagnosticAdapter/DiagnosticListenerExtensions.cs]: Cool beans 👍

**Comment by rynowak** [comment]
:shipit:
